delta/plugins/archive/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/archive/model/ArchiveReference.scala

```diff
@@ -74,4 +79,57 @@ object ArchiveReference {
       ref.original.toString + project.toString + path.toString + representation.toString
   }
 
+  implicit final val archiveReferenceJsonLdDecoder: JsonLdDecoder[ArchiveReference] = {
+    // custom decoder for resource ref to rev and tag references from separate fields
+    implicit val resourceRefDecoder: JsonLdDecoder[ResourceRef] =
+      (cursor: ExpandedJsonLdCursor) =>
+        for {
+          iri    <- cursor.downField(nxv + "resourceId").get[Iri]
+          revOpt <- cursor.downField(nxv + "rev").get[Option[Long]]
+          tagOpt <- cursor.downField(nxv + "tag").get[Option[TagLabel]]
+          value  <- (revOpt, tagOpt) match {
+                      case (None, None)      => Right(ResourceRef.Latest(iri))
+                      case (Some(rev), None) => Right(ResourceRef.Revision(iri, rev))
+                      case (None, Some(tag)) => Right(ResourceRef.Tag(iri, iri, tag))
+                      case _                 =>
+                        Left(ParsingFailure("An archive resource reference cannot use both 'rev' and 'tag' fields."))
+                    }
+        } yield value
+
+    // path decoder that checks for absolute paths
+    implicit val absolutePathDecoder: JsonLdDecoder[Path] = {
+      val pathDecoder: JsonLdDecoder[Path] = (cursor: ExpandedJsonLdCursor) =>
+        cursor.getValueTry[Path](str => Paths.get(str))
+      pathDecoder.flatMap { path =>
+        if (path.isAbsolute) Right(path)
+        else Left(ParsingFailure(s"The archive reference path '$path' is not absolute."))
+      }
+    }
+
+    val resourceReferenceDecoder: JsonLdDecoder[ResourceReference] =
```

Comment:
This can be done automatically using `deriveConfigJsonLdDecoder` as for storages or elasticsearch views. it is a trait with different implementations by a discriminator

Reply:
because of the custom decoding of `resourceRef` derivation cannot be used here; that's what I tried first.

Reply:
The resourceRef decoding is already provided, so what's the problem?

Reply:
if you have the following json:
```
{
  "@type": "Resource"
  "resourceId": "http://localhost/id",
  "rev": 1,
  "path": "/id.json"
}
```
you have to decode the same json twice:
1. to get the `ResourceRef` from the root
2. to get the `ResourceReference` from the root

The generic derivation would expect something like the following to work:
```
{
  "@type": "Resource"
  "ref": {
    "resourceId": "http://localhost/id",
    "rev": 1
  },
  "path": "/id.json"
}
```

Reply:
that's why we usually have a different case class to deserialize Json-LD than to actually use it for command/events(like in elasticsearch), it would probably be more readable.